### ISSUE TITLE
Stripe ACH Integration in Beta

### ISSUE THREAD

**Issue by dsuguy2007** [drastik/com.drastikbydesign.stripe#124]
Stripe is beta testing ACH and will be rolling it out in the near future.  Please integrate that feature.

**Comment by dsuguy2007** [comment]
I got a response from Stripe and the ACH feature is ready to go they will be making it public soon. 
 after lots of testing and iteration, we’re delighted to give you a preview of the product that we'll be announcing soon.

Our ACH support works with all Stripe products: You can create one-off transactions or charge for subscriptions, as Tableau, Slack, or Digital Ocean do. Companies using Connect, like Fancy or Tilt, can accept ACH payments on behalf of other customers. As you’d expect, reporting for ACH payments lives alongside credit card payments in the Dashboard.

Handling ACH payments can be more complex than charging credit cards since the transaction amounts are often larger and require trickier authorization. So, our ACH support comes with built-in micro-deposit and instant verification support.

We support adding bank accounts in two ways: via micro-depots or by instantly verifying the customer with Plaid. For micro-deposits, Stripe automatically makes two small deposits when you create a bank account token and customers must confirm that they own the bank account by confirming these two amounts.

We also worked with Plaid on a Connect application that offers instant verification and a simpler checkout process for ACH payments.

Your customers can securely sign into their bank account and be instantly verified—no digging around for routing numbers or waiting several days for micro-deposit amounts. (This process also prevents most of the common errors of mistyping bank account numbers or incorrect routing numbers.)

**Comment by dsuguy2007** [comment]
And it's now public: documentation at:
https://stripe.com/docs/guides/ach

**Comment by jmcclelland** [comment]
We just had a customer ask for ACH support in CiviCRM Stripe. 

I'm not quite sure how it would be implemented.

Stripe seems to suggest that it would be our responsibility to have the payer confirm the micro payments several days after making a contribution. (Or use plaid - which I'm also a bit vague on.)

I see that IATS has implemented ACH for CiviCRM. However, the IATS [documentation on ach](http://home.iatspayments.com/developers/sample-code/process-transactions) doesn't say anything about customers verifying it.

**Comment by jmcclelland** [comment]
More detailed information is here:

https://stripe.com/docs/guides/ach

Seems like integrating with Plaid is the only reasonable way to have a smooth verification process. If you don't use Plaid, or if someone wants to make a payment using a bank that is not supported by plaid, then the first payment will generate two mirco-payments. And, several days later, the person who is trying to make the payment, will have to check their account statement and then come back and confirm the amounts of the payments.

**Comment by dsuguy2007** [comment]
Right, which I talked with Stripe about and complained that it was not a very customer friendly method. Also Plaid charges extra fees as well. So it looks like ACH with Stripe is a big FAIL!!

**Comment by drastik** [comment]
:-1:

**Comment by lahari509** [comment]
when stripe deposits the micro deposits to the customer bank . how we know that customer verifies his bank account
